### DISCUSSION TITLE
Containers have disappeared

### DISCUSSION THREAD

**Discussion by cueup** [caprover/caprover#2048] [General]
@githubsaturn - I've messed up. 

I was working on my production server trying to setup a sendmail and something strange happened. 

It wasn't working and out of frustration I ran the following commands to remove and start again:
sudo apt purge sasl2-bin
sudo apt purge postfix
sudo apt-get purge sendmail*

Then when I tried accessing some of my application I got a 502 error. Funny thing is one of my application (and DBs) are working fine and is still listed in the containers when i run docker ps. 

The three apps that are no longer visible in when i run docker ps and there in the docker images repo. The apps are also still present in the caprover dashboard. 

When I check the syslog I get the following errors:

Apr 26 13:20:06 ip-MY-INTERNAL-IP dockerd[679]: time="2024-04-26T13:20:06.416834394Z" level=error msg="pulling image
 failed" error="pull access denied for img-captain-MY_APP, repository does not exist or may require 'docker l
ogin': denied: requested access to the resource is denied" module=node/agent/taskmanager node.id=xn3gxy6yd5myopyn
col62jx1b service.id=pks36txil9mqfqltot5b23tvz task.id=aqzcy4acffyig3w8sq0ua9m0d

**I wanted to find out if there is a way to redeploy from the last deployed image and have it link back to the same caprover app?** 

I've tried docker run MY_APPS_IMAGE name but it just creates a new container and does not link back to the dashboard app. 

Unfortunately for me, I did not turn of server backup.

**Comment by githubsaturn** [comment]
Are these apps (that don't work) the apps that you deployed from source code? Or one click apps?

**Reply by cueup** [reply]
I managed to resolve this by running IMAGENAME:latest on  the deployment tab  Method 4: Deploy plain Dockerfile
